fearless/Modules/NewWallet/ChainAccountBalanceList/ChainAccountBalanceListInteractor.swift

```diff
@@ -99,10 +110,69 @@ final class ChainAccountBalanceListInteractor {
         }
     }
 
-    private func provideCurrency() {
-        let currentCurrency = settingsManager.selectedCurrency
-        self.currentCurrency = currentCurrency
-        presenter?.didReceiceCurrency(currentCurrency)
+    private func subscribeToFiats() {
+        fiatInfoProvider = nil
+
+        guard let fiatUrl = ApplicationConfig.shared.fiatsURL else { return }
+        fiatInfoProvider = jsonDataProviderFactory.getJson(for: fiatUrl)
+
+        let updateClosure: ([DataProviderChange<[Currency]>]) -> Void = { [weak self] changes in
+            guard let strongSelf = self else { return }
+            if var result = changes.reduceToLastChange() {
+                guard
+                    let selectedCurrencyIndex = result.firstIndex(where: { $0.id == strongSelf.currency.id })
+                else {
+                    return
+                }
+
+                result[selectedCurrencyIndex] = strongSelf.currency
```

Comment:
please pass selected currency to presenter and viewmodelfactory instead of modifying response